src/main/java/seedu/address/model/person/OrganExpiryDate.java

```diff
@@ -0,0 +1,95 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a Donor organ's expiry date in ORGANice.
+ * Guarantees: immutable; is valid as declared in {@link #isValidExpiryDate(String)} (String)}
+ */
+public class OrganExpiryDate {
+
+    public static final Map<Integer, Integer> NUMBER_OF_DAYS_IN_MONTHS = new HashMap<>() {{
+            put(1, 31);
+            put(2, 29);
+            put(3, 31);
+            put(4, 30);
+            put(5, 31);
+            put(6, 30);
+            put(7, 31);
+            put(8, 31);
+            put(9, 30);
+            put(10, 31);
+            put(11, 30);
+            put(12, 31);
+        }};
+    public static final String MESSAGE_CONSTRAINTS = "Organ's expiry date must be in the format DD-MMM-YYYY"
+            + " and is after current date. An example will be 27-Jan-2020";
+
```

Comment:
That being said, yoru current setup is stricter than what the DTF offers so that might be a good thing. Your call to make here

Reply:
Aiight okay I realise its probably better if you use DateFormat instead of DateTimeFormatter, since there is no time element. Also, there is a "strict" parsing mode that throws exception when nonsense like 31 Feb is being parsed. More on that [here](https://stackoverflow.com/questions/30574629/prevent-invalid-date-from-getting-converted-into-date-of-next-month-in-jdk6)